src/SciMLBase.jl

```diff
@@ -367,6 +368,16 @@ $(TYPEDEF)
 """
 abstract type AbstractSDDEIntegrator{Alg, IIP, U, T} <: DEIntegrator{Alg, IIP, U, T} end
 
+# Discretizers
+"""
+$(TYPEDEF)
+
+## Type parameters:
+- `N` The number of discretized independent variables.
+- `M` The number of discretized dependent variables.
```

Comment:
why do these need to be type parameters?

Why is this not an abstract array with time / without time?

Reply:
What should be an abstract array? They don't need to be type parameters, I just thought this made sense to expose.